admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,511 @@
+
+{product-title} 
+might scheduling pods to the bad nodes.
+
+{product-title} supports the following problem daemons. Installation of the daemone is beyond the scope of this document.  
+
+|===
+| Problem Daemon | NodeCondition | Description
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json[KernelMonitor]
+| KernelDeadlock
+| A system log monitor monitors kernel log and reports problem according to predefined rules.
+
+| link:https://github.com/abrt/abrt/wiki[AbrtAdaptor]
+| None
+| Monitor ABRT log messages and report them further. ABRT (Automatic Bug Report Tool) is health monitoring daemon able 
+to catch kernel problems as well as application crashes of various kinds occurred on the host.
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor that checks various node problems with xref:admin-guide-node-problem-detector-monitor[custom, user-defined scripts].
+
+|===
+
+//https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#node-problem-detector
+The node problem detector supports file-based kernel logging. However, it is easy to extend it to xref:admin-guide-node-problem-detector-log[support other log format].
+
+The xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[advanced installation] creates 
+a node problem detector daemonset by default and creates a project for the detector to run, called `openshift-node-problem-detector`
+
+== Using the Node Problem Detector
+
+Verify that the node problem detector is running:
+
+[source,bash]
+----
+oc logs <pod_name>
+----
+
+For example:
+
+[source,bash]
+----
+oc logs node-problem-detector-c6kng
+I0416 23:22:00.641354       1 log_monitor.go:63] Finish parsing log monitor config file: {WatcherConfig:{Plugin:journald PluginConfig:map[source:kernel] LogPath:/host/log/journal Lookback:5m} BufferSize:10 Source:kernel-monitor DefaultConditions:[{Type:KernelDeadlock Status:false Transition:0001-01-01 00:00:00 +0000 UTC Reason:KernelHasNoDeadlock Message:kernel has no deadlock}]
+----
+
+To use the node problem detector, use the `oc get` command with a resource:
+
+[source,bash]
+----
+oc get node registry-router-1 -o yaml | grep -B5 KernelDeadlock
+
+ - lastHeartbeatTime: 2018-04-17T08:01:44Z
+    lastTransitionTime: 2018-04-17T08:01:43Z
+    message: task docker:1234 blocked for more than 700 seconds.
+    reason: DockerHung
+    status: "True"
+----
+
+[[admin-guide-node-problem-detector-run]]
+== Reinstalling the Node Problem Detector
+
+If for any reason, you need to re-create the xref:../dev_guide/daemonsets#dev-guide-daemonsets[daemonset] object, run the following Ansible playbook:
+
+[source,bash]
+----
+ansible-playbook ~/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
```

Comment:
Since the local checkout location of the openshift-ansible repo will vary from person to person, I'd say that we should leave the `~/` off. Or maybe we should refer to the one installed by the openshift-ansible-playbooks RPM on a node. Here's the location of the file from the RPM if you want to use that:

```
/usr/share/ansible/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
```

Reply:
I think my earlier comment is still applicable. This should be `ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml` and the text above it should mention that the file comes from the `openshift-ansible-playbooks` RPM.